iothub/service/src/Configurations/ConfigurationsClient.cs

```diff
@@ -0,0 +1,422 @@
+                    Logging.Exit(this, $"Deleting configuration: {configurationId}", nameof(DeleteAsync));
+            }
+        }
+
+        /// <summary>
+        /// Deletes a previously registered device from the system.
+        /// </summary>
+        /// <param name="configuration">The configuration being deleted.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the provided configuration is null.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task DeleteAsync(Configuration configuration, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Deleting configuration: {configuration?.Id}", nameof(DeleteAsync));
+            try
+            {
+                Argument.RequireNotNull(configuration, nameof(configuration));
+                cancellationToken.ThrowIfCancellationRequested();
+                if (string.IsNullOrWhiteSpace(configuration.ETag)) 
+                {
+                    throw new ArgumentException(ApiResources.ETagNotSetWhileDeletingConfiguration);
+                }
+                await DeleteAsync(configuration.Id, configuration, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(DeleteAsync)} threw an exception: {ex}", nameof(DeleteAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Deleting configuration: {configuration?.Id}", nameof(DeleteAsync));
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="configurationId"></param>
+        /// <param name="eTagHolder"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task DeleteAsync(string configurationId, IETagHolder eTagHolder, CancellationToken cancellationToken)
```

Comment:
Customer's aren't expected to know about this IETagHolder type. The only two ```DeleteAsync``` APIs we should have are 
```csharp
DeleteAsync(string configurationId, CancellationToken cancellationToken = default);

DeleteAsync(Configuration configuration, CancellationToken cancellationToken = default);
```

Reply:
Move the implementation within this method to the ```DeleteAsync(Configuration configuration, CancellationToken cancellationToken = default``` method